### DISCUSSION TITLE
Ginlong Solis Documentation

### DISCUSSION THREAD

**Discussion by wills106** [wills106/homeassistant-solax-modbus#416] [Ideas]
I have started to document Solis Inverters in the [Wiki](https://github.com/wills106/homeassistant-solax-modbus/wiki/Ginlong-Solis-Modes-of-Operation)

Can any Solis Owners help filling in the missing gaps. If you could post descriptions below I can start adding it into the Wiki.

Also any completely new sections are welcome.

**Comment by alienatedsec** [comment]
**Auto-Mode**
https://github.com/wills106/homeassistant-solax-modbus/wiki/Ginlong-Solis-Modes-of-Operation#auto-mode---no-grid-charge

```
It's also known as a `Self-Use` mode on the inverter. It will use the PV generation to charge the battery where any excess is to be exported to the grid. In this mode, the inverter is to provide power according to its capability, and anything outside of its limits is to be provided from the grid.

The below is based on the inverter `RHI-6K-48ES-5G` - 6kW model with the battery.

**Scenario 1:**
House Load: 7000W
PV Generation: 5000W
Inverter is to provide 5000W from the PV, 1000W from the battery and 1000W from the grid.

**Scenario 2:**
House Load: 1000W
PV Generation: 300W
Inverter is to provide 300W from the PV and the rest from the battery.
```

@wills106 any other format or more examples? I can add them whenever needed

[EDIT] we need two scenarios as it will differ per mode.

**Reply by wills106** [reply]
So Auto/Self-Use is basically the same as the SolaX mode here https://github.com/wills106/homeassistant-solax-modbus/wiki/SolaX-Gen2-&-Gen3-Modes-of-operation#self-use

I don't mind on formatting, just need get some content in there especially for new users.
For the SolaX I started with what's in the manual and expanded on it where needed.

**Reply by wills106** [reply]
So Scenario 1 = Auto
Scenario 2 = Auto - No Grid Charge

With scenario 2 what happens if the house load can't be met by PV and Battery?

**Reply by alienatedsec** [reply]
Looking at SolaX's description of `Self-Use`, it matches the `Auto-Mode` of Solis. My description above was for the same mode but provided some scenarios for that mode. Also, in `Auto - No Grid Charge` mode there will be another scenario. Not sure if it is worth me reinventing the whole thing.

**Reply by alienatedsec** [reply]
Page 38 onwards

https://www.ginlong.com/uploads/file/Solis_Manual_RHI(3-6)K_5G_ENX_V1,5.pdf

**Reply by wills106** [reply]
If I can just copy any of the SolaX descriptions for any of the Solis ones just let me know what to copy over and then we can add in any missing bits.

**Reply by alienatedsec** [reply]
The problem is that setting multiple settings on the inverter could be equivalent to setting a single register value in the integration. Many are not described or are very vague in description.

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/73167064/a578c034-be41-485a-96ff-d26e6cc2b503)

**Reply by alienatedsec** [reply]
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/73167064/c57a7563-81d2-48ac-a06b-c95217aa0764)

**Reply by Cabocab** [reply]
For typical use, I've found that there is functionally no difference between Auto Mode Grid Charge/No Grid Charge.  The only difference was when I had 2 inverters in parallel (1 hybrid and 1 grid tie).  When the generation is high enough that the entire system is exporting to grid, the hybrid will try to charge the batteries from grid until export is 0.   In this case, the hybrid can potentially charge the battery with power higher than the pv panels installed.

Let's say the panels connected to the hybrid  is generating 1 kW and all of it is used to charge the battery.  If you have another source that causes the system to export (another inverter, or other renewable energy source) let's say 500 W, the hybrid will try to "charge from grid" to zero out the export.  In this case the hybrid will charge the batteries at 1.5 kW (1 kW from panels, 500 W from "grid") to maintain the zero export.  If grid charge is off, 1 kW from panels will charge the battery, and 500 W will be exported.

**Reply by wills106** [reply]
So should this really be called Auto Mode - No Grid Export?

I forget some markets discourage you from exporting.

**Reply by Cabocab** [reply]
As @alienatedsec mentioned, Solis calls this "self-use" mode.  It maximizes self-use (including charging batteries), but when the batteries are full, it will still export any excess generation.

There is a separate mode/setting that completely disables/limits grid export (Backflow Power Switch in the integration) but it has some weird interactions with the other modes, although I am not sure if this is weird interaction is just with my inverter model.

**Reply by wills106** [reply]
Yea Self-Use makes more sense as it matches up with SolaX Inverters etc. I was just going off what's in the Modbus Docs, which don't always match what's on the Inverter LCD / Cloud Portal's.

I can update the naming on the next release / beta release.
Are there any other names that need correcting? Although I'll have to state them as breaking changes as people might be using them in automations etc.

**Reply by wills106** [reply]
1: "Selfuse - No Grid Export",
3: "Timed Charge/Discharge - No Grid Export",
33: "Selfuse",

Or

1: "Self-Use - No Grid Export",
3: "Timed Charge/Discharge - No Grid Export",
33: "Self-Use",

Also is "No Grid Export" misleading? Is it possible for it to still export even when the Battery if full?

**Reply by Cabocab** [reply]
I think it should be No Grid Charging instead of No Grid Export as this disables the Grid Charging flag.
